### PR TITLE
Add pagination to SimpleHttpOperator

### PR DESCRIPTION
**Pull request by Joffreybvn**
Hello,

This PR implements support for paginated API in the `SimpleHttpOperator`. No breaking changes, but adds a lot of code into this Operator. I can also propose a PR with a new _PaginatedHttpOperator_ if better ?

#### Use case:
I have been facing a very specific use-case where the API:
- Returns data (as JSON) in a paginated way
- Do not provide any way to know in advance how much pages / entries will be returned
- Provides a "next_page_cursor" attribute, which is a random id, to do paginated calls

_Dynamic Tasks?_  It works like a for loop. In this case, knowing in advance the number of pages means calling the API two times. Which is inefficient, and do not guarantee all pages are traversed during the second call.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by jens-scheffler-bosch** [comment]
Thanks for the contribution! I was not aware of cases where pagination is used until I saw this PR. I don't know how common this is in API Calls and might be very specific.
As the HTTP operator is very basic and used in many use cases I am not sure whether a pagination function really is best to be suited in the base operator or as proposed an extended Operator inheriting from the SimpleHttpOperator would be better.

Also - to be open - I am not sure whether such function is so common that it makes sense to add it into the code package. I believe (but this is my personal opinion - other opinions welcome!) that this could also be very good placed in a contributor package to keep the basic operator "simple". Python is extensible and the standard operator can easily be used and extended per customer project.

(I am not a committer so my opinion only counts a bit, would be good to get other maintainers opinion on this contribution - also if not in code where such extension could be best placed in.)

File: airflow/providers/http/operators/http.py [hunk 1]
```diff
@@ -49,6 +50,12 @@ class SimpleHttpOperator(BaseOperator):
     :param data: The data to pass. POST-data in POST/PUT and params
         in the URL for a GET request. (templated)
     :param headers: The HTTP headers to be added to the GET request
+    :param pagination_function: A callable that generate the parameters used
+        to call the API again. Typically used when the API is paginated and
+        returns a cursor, a next page id, or a next page URL. When provided,
+        the Operator will call the API repeatedly until this function returns
+        None. Also, by default, the result of the Operator will be a list of
+        Response.text objects.
     :param response_check: A check against the 'requests' response object.
```
Comment (utkarsharma2) [inline airflow/providers/http/operators/http.py:61]: Just a minor thing :)
The docstring for `response_check` and `response_filter` should reflect the fact that the response could be a list of responses when the `pagination_function` is passed.

**Review by utkarsharma2 — APPROVED** [review]

File: airflow/providers/http/operators/http.py (hunk 1 above)
Comment (Joffreybvn) [reply airflow/providers/http/operators/http.py:61]: Updated it, thanks !

**Comment by hussein-awala** [comment]
From his name `SimpleHttpOperator` and the description:
```
Calls an endpoint on an HTTP system to execute an action.
```
I see that it should be simple and just used to execute actions and not load big payloads which need pagination.

IMHO creating a new operator is better for this feature, and try to keep it generic, like `HttpOperator` and maybe a subclass of the first one if possible

**Comment by Joffreybvn** [comment]
Alright, thanks for reviews ! I'll create a new PR with a Paginated one, based on the HttpOperator

**Pull request closed without merging**